src/transformStateWithClones.js

```diff
@@ -7,7 +7,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
-}
+  const stateHistory = [];
+  const newState = { ...state };
+
+  for (const keys in actions) {
```

Comment:
[CODE STYLE]: Use a `switch` statement if you have a limited amount of conditions. This will make the code cleaner and more readable.